### PR TITLE
ISPN-13099 Support authentication for all commands in CLI

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-13099

The `shortName` was removed. They are matching some in `org.infinispan.cli.commands.Benchmark`

### CODE REVIEW

File: cli/src/main/java/org/infinispan/cli/commands/CliCommand.java [hunk 1]
```diff
@@ -1,24 +1,79 @@
...
+import org.wildfly.security.keystore.KeyStoreUtil;
+import org.wildfly.security.provider.util.ProviderUtil;
 
 /**
  * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
  * @since 10.0
  **/
 public abstract class CliCommand implements Command<ContextAwareCommandInvocation> {
 
+   @Option(description = "A connection URL. Use '-' to connect to http://localhost:11222")
```
Comment (oraNod) [inline cli/src/main/java/org/infinispan/cli/commands/CliCommand.java:33]:
```suggestion
   @Option(description = "Specifies the URL for the server connection. Use '-' to connect to http://localhost:11222")
```

File: cli/src/main/java/org/infinispan/cli/commands/CliCommand.java [hunk 2]
```diff
@@ -1,24 +1,79 @@
...
+   String connect;
+
+   @Option(hasValue = false, description = "Whether to trust all certificates", name = "trustall")
```
Comment (oraNod) [inline cli/src/main/java/org/infinispan/cli/commands/CliCommand.java:36]:
```suggestion
   @Option(hasValue = false, description = "Trusts all certificates as valid server identities.", name = "trustall")
```

File: cli/src/main/java/org/infinispan/cli/commands/CliCommand.java [hunk 3]
```diff
@@ -1,24 +1,79 @@
...
+   boolean trustAll;
+
+   @Option(completer = FileOptionCompleter.class, name = "truststore", description = "A truststore to use when connecting to SSL/TLS-enabled servers")
```
Comment (oraNod) [inline cli/src/main/java/org/infinispan/cli/commands/CliCommand.java:39]:
```suggestion
   @Option(completer = FileOptionCompleter.class, name = "truststore", description = "Specifies a trust store that contains SSL/TLS certificates to verify server identities.")
```

File: cli/src/main/java/org/infinispan/cli/commands/CliCommand.java [hunk 4]
```diff
@@ -1,24 +1,79 @@
...
+   Resource truststore;
+
+   @Option(name = "truststore-password", description = "The password for the truststore")
```
Comment (oraNod) [inline cli/src/main/java/org/infinispan/cli/commands/CliCommand.java:42]:
```suggestion
   @Option(name = "truststore-password", description = "Specifies the password for the trust store.")
```

**Review by oraNod — CHANGES_REQUESTED** [review]

**Review by oraNod — APPROVED** [review]
LGTM

**Comment by pruivo** [comment]
-1 for this PR for 2,5 reasons
* commands that don't need a "connection" (example `user`, `patch`, `cd`, etc) are affected as well. I mean, this makes no sense
```
[pedro-desktop-51675@cluster//containers/default]> disconnect --connect=http://127.0.0.1:11222
Username: pedro
Password: *****
[disconnected]> 
```
* if `AUTOCONNECT_URL`is set, it creates a new connection for each command (even `cd`)
* the issue is in the parser that ignores the args if you use an option

**Comment by diegolovison** [comment]
* commands that don't need a "connection"
```
[disconnected]> connect
Username: admin
Password: *****
[dlovison-mac-43286@cluster//containers/default]> disconnect
[disconnected]> 
```
it works without the `connect` attribute

**Comment by tristantarrant** [comment]
I have spoken to Stale on how to implement this properly (i.e. multiple commands on the CLI). It requires some interaction with the parser

**Comment by tristantarrant** [comment]
My proposed solution would allow for things like:
`cli connect -u username -p password shutdown`

**Comment by diegolovison** [comment]
Why not?
The `--connect` should go in the end. Is this the problem?
```
(base) dlovison@dlovison-mac bin % ./cli.sh shutdown server --connect http://admin:admin@127.0.0.1:11222 
(base) dlovison@dlovison-mac bin % 
```

**Comment by pruivo** [comment]
the problem, IMO of course, is that it causes confusion when run in interactive since the autocompletion adds the `--connect` when you look for help

**Comment by tristantarrant** [comment]
And, to make matters worse: graalvm's native-image completely messes up inherited fields, making all this pointless

**Comment by tristantarrant** [comment]
I'd like to close this since we'll move away from aesh towards picocli

**Pull request closed without merging**